src/univariate/discrete/bernoulli.jl

```diff
@@ -113,44 +113,41 @@ rand(rng::AbstractRNG, d::Bernoulli) = rand(rng) <= succprob(d)
 
 #### MLE fitting
 
-struct BernoulliStats <: SufficientStats
-    cnt0::Float64
-    cnt1::Float64
-
-    BernoulliStats(c0::Real, c1::Real) = new(Float64(c0), Float64(c1))
+struct BernoulliStats{C<:Real} <: SufficientStats
+    cnt0::C
+    cnt1::C
 end
 
+BernoulliStats(c0::Real, c1::Real) = BernoulliStats(promote(c0, c1)...)
+
 fit_mle(::Type{<:Bernoulli}, ss::BernoulliStats) = Bernoulli(ss.cnt1 / (ss.cnt0 + ss.cnt1))
 
 function suffstats(::Type{<:Bernoulli}, x::AbstractArray{T}) where T<:Integer
-    n = length(x)
     c0 = c1 = 0
-    for i = 1:n
+    for i = eachindex(x)
         @inbounds xi = x[i]
```

Comment:
A bit simpler:
```suggestion
    for xi in x
```

Reply:
The verbose version is _slightly_ faster...
Similarly,
```julia
for i = eachindex(x, w)
        @inbounds xi = x[i]
        @inbounds wi = w[i]
```
is faster than
```julia
for (xi, wi) in zip(x, w)
```
I can change it though, in which case one could also remove the `::AbstractArray` bound to allow for other iterators.

Reply:
We don't want to remove the `::AbstractArray` types in the second case (at least not currently), it's only about not breaking existing code. `eachindex(x, w)` is too restrictive.